### ISSUE TITLE
defaultKey function fails when directories are passed

### ISSUE THREAD

**Issue by michaelherold** [jgable/gulp-cache#42]
I'm encountering an exception when trying to start my gulp project with gulp-cache-0.2.7. The gulpfile.js works with gulp-cache-0.2.4.

You can see the exception I'm encountering in [this gist](https://gist.github.com/michaelherold/dc145c3eaac4349c0ad7).

Please let me know if you need more information. I'm happy to give you any info you need, it's just that I'm not particularly familiar with what's useful for debugging gulp scripts.

**Comment by jgable** [comment]
I think you've discovered the same issue as #41.  The crux is that directories are causing the `defaultKey` function to fail.

A workaround is described by @pavangupta in https://github.com/google/web-starter-kit/pull/636/files.  We should have a fix in before the end of the day though.

**Comment by michaelherold** [comment]
Yep, that sounds like the same problem. Thanks for the quick triage. I'm looking forward to the fix and will happily test it when that fix is released.

**Comment by shinnn** [comment]
@michaelherold Could you try using this version?

```sh
npm  install -D jgable/gulp-cache#4f8e1dc5dce8535ed8dca30cb9923658e1cdf9de
```

**Comment by shinnn** [comment]
If the problem for @michaelherold is fixed, I'll release a new minor-updated version.

**Comment by michaelherold** [comment]
The hash you gave me wouldn't install, but I did use 031a379cd47a59c4ad300e9b2328fe4f6cd5d830 (which is the merge commit) and it successfully runs.

:+1:

**Comment by jgable** [comment]
Thanks for checking it out @michaelherold :+1:

**Comment by shinnn** [comment]
> The hash you gave me wouldn't install

Ohhhhh, sorry I deleted the branch after the PR is merged. Thanks!

**Comment by shinnn** [comment]
> I'll release a new minor-updated version.

I was wrong. we should release a patch-updated version instead.

**Comment by shinnn** [comment]
Released v0.2.8.

**Comment by pavangupta** [comment]
Woot!  Nicely done, @shinnn!
